modules/aruco/src/aruco.cpp

```diff
@@ -1330,6 +1329,33 @@ void GridBoard::draw(Size outSize, OutputArray _img, int marginSize, int borderB
 
 
 /**
+*/
+Ptr<Board> Board::create(InputArrayOfArrays objPoints, Ptr<Dictionary> &dictionary, InputArray ids) {
+
+    Mat ids_mat = ids.getMat();
+    std::vector<int> ids_vector;
+    for (int i = 0; i < ids_mat.rows; i++) {
+        ids_vector.push_back(ids_mat.at<int>(i));
```

Comment:
I think you can just do `ids.copyTo(ids_vector)` here. Or even better: `ids.copyTo(res->ids)`